### PR TITLE
Adjust version check for TokensInvalidationResult

### PR DESCRIPTION
**Pull request by jkakavas**
Now that the relevant changes are backported to `7.x`, `7.1` nodes in
BWC test will not expect to read/write the attempt counter so we
can adjust the version check on serialization/deserialization.
This change also enables the tests that have been muted while this
was backported

### CODE REVIEW

**Comment by jkakavas** [comment]
@elasticmachine run elasticsearch-ci/bwc elasticsearch-ci/default-distro

File: x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/authc/TokenService.java [hunk 1]
```diff
@@ -1348,7 +1348,7 @@ private String getFromHeader(ThreadContext threadContext) {
      * itself for versions after 7.0.0
```
Comment (albertzaharovits) [inline x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/authc/TokenService.java]: nit s/7.0.0/7.1.0

**Comment by albertzaharovits** [comment]
Found two more places that need to have the version 7.1 instead of 8.0, in `TokenService`:

in `checkLenientlyIfTokenAlreadyRefreshed`:
```
        Version authVersion = Version.fromId(version);
        final Boolean refreshed = (Boolean) refreshTokenSrc.get("refreshed");
        if (refreshed) {
            if (authVersion.onOrAfter(Version.V_8_0_0)) {
                final Long refreshedEpochMilli = (Long) refreshTokenSrc.get("refresh_time");
                final Instant refreshTime = refreshedEpochMilli == null ? null : Instant.ofEpochMilli(refreshedEpochMilli);
                final String supersededBy = (String) refreshTokenSrc.get("superseded_by");
                if (supersededBy == null) {
```
and in `eligibleForMultiRefresh`:
```
        final Long refreshedEpochMilli = (Long) refreshTokenSrc.get("refresh_time");
        final Instant refreshTime = refreshedEpochMilli == null ? null : Instant.ofEpochMilli(refreshedEpochMilli);
        final String supersededBy = (String) refreshTokenSrc.get("superseded_by");
        return authVersion.onOrAfter(Version.V_8_0_0)
            && supersededBy != null
            && refreshTime != null
            && refreshRequested.isBefore(refreshTime.plus(30L, ChronoUnit.SECONDS))
            && refreshRequested.isAfter(refreshTime.minus(30L, ChronoUnit.SECONDS));
    }
```

**Review by albertzaharovits — APPROVED** [review]
LGTM

**Pull request merged**